src/Patterns/ProductUpdater.php

```diff
@@ -0,0 +1,361 @@
+		);
+
+		$dummy_products_to_update = [];
+		foreach ( $dummy_products as $dummy_product ) {
+			$current_product_hash     = $this->get_hash_for_product( $dummy_product );
+			$ai_modified_product_hash = $this->get_hash_for_ai_modified_product( $dummy_product );
+
+			$date_created  = $dummy_product->get_date_created()->date( 'Y-m-d H:i:s' );
+			$date_modified = $dummy_product->get_date_modified()->date( 'Y-m-d H:i:s' );
+
+			$timestamp_created  = strtotime( $date_created );
+			$timestamp_modified = strtotime( $date_modified );
+
+			$dummy_product_not_modified = abs( $timestamp_modified - $timestamp_created ) < 60;
+
+			if ( $current_product_hash === $ai_modified_product_hash || $dummy_product_not_modified ) {
+				$dummy_products_to_update[] = $dummy_product;
+			}
+		}
+
+		if ( empty( $dummy_products_to_update ) ) {
+			return true;
+		}
+
+		$ai_selected_products_images = $this->get_images_information( $vertical_images );
+		$products_information_list   = $this->assign_ai_selected_images_to_dummy_products_information_list( $ai_selected_products_images );
+
+		$responses = $this->generate_product_content( $products_information_list );
+
+		foreach ( $responses as $key => $response ) {
+			if ( is_wp_error( $response ) ) {
+				return $response;
+			}
+
+			if ( empty( $response ) ) {
+				return new \WP_Error( 'empty_response', __( 'The response from the AI service was empty.', 'woo-gutenberg-products-block' ) );
+			}
+
+			if ( ! isset( $response['completion'] ) ) {
+				continue;
+			}
+
+			$product_content = json_decode( $response['completion'], true );
+
+			if ( is_null( $product_content ) ) {
+				continue;
+			}
+
+			// This is required to allow the usage of the media_sideload_image function outside the context of /wp-admin/.
+			// See https://developer.wordpress.org/reference/functions/media_sideload_image/ for more details.
+			require_once ABSPATH . 'wp-admin/includes/media.php';
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+			require_once ABSPATH . 'wp-admin/includes/image.php';
+
+			$i = 0;
+			foreach ( $dummy_products_to_update as $dummy_product ) {
+				$this->update_product_content( $dummy_product, $product_content[ $i ] );
+				++$i;
+			}
+		}
+
+		$update_option = update_option( 'last_business_description_with_ai_content_generated', $business_description );
+
+		if ( ! $update_option ) {
+			return new \WP_Error( 'update_option_failed', __( 'The option last_business_description_with_ai_content_generated could not be updated.', 'woo-gutenberg-products-block' ) );
+		}
+
+		return $update_option;
+	}
+
+	/**
+	 * Creates a new product and assigns the _headstart_post meta to it.
+	 *
+	 * @return bool|int
+	 */
+	public function create_new_product() {
+		$product = new \WC_Product();
+
+		$product->set_name( 'My Awesome Product' );
+		$product->set_status( 'publish' );
+		$product->set_description( 'Product description' );
+		$product->set_price( 25 );
+		$product->set_regular_price( 25 );
```

Comment:
Non-blocker but just a thought. Would be great to randomise these values and for _some_ products add sale prices for variety.

Reply:
I've updated the code to randomise the price as suggested: https://github.com/woocommerce/woocommerce-blocks/pull/11155/commits/3da76eb5ef1ea018cb9f6de71a1b5a2cfabba7cd